linux_kernel_modules/cypwifi/scripts/cywifi.sh

```diff
@@ -44,10 +45,20 @@ cy_wifi_init () {
 	insmod  /legato/systems/current/modules/cfg80211.ko
 }
 cy_wifi_start() {
-	# Let's bring the Cypress chip out of reset - see Note above not needing now
-	#echo ${WL_REG_ON_GPIO} > /sys/class/gpio/export
-	#echo out  > /sys/class/gpio/gpio${WL_REG_ON_GPIO}/direction
-	#echo 1  > /sys/class/gpio/gpio${WL_REG_ON_GPIO}/value
+	# Let's bring the Cypress chip out of reset
+	if [ "${CF3}" = "mdm9x28" ] ; then
+		if [ ! -d "/sys/class/gpio/gpio${WL_REG_ON_GPIO}" ] ; then
+			echo ${WL_REG_ON_GPIO} > /sys/class/gpio/export
+			echo out  > /sys/class/gpio/gpio${WL_REG_ON_GPIO}/direction
+		fi
+		WL_REG_ON_VALUE=`cat /sys/class/gpio/gpio${WL_REG_ON_GPIO}/value`
+		if [ $WL_REG_ON_VALUE -eq 0 ] ; then
+			echo 1  > /sys/class/gpio/gpio${WL_REG_ON_GPIO}/value
+			sleep 2
+			modprobe sdhci-msm
+			sleep 2
```

Comment:
Is there a more deterministic way that this can be done?  Sleeps generally create race conditions.

Reply:
I agree wholeheartedly and am bothered by the lack of handshaking. I had tried this previously on the WP85 and the WL_HOST_WAKE did not work. On WP76 Yellow I just tried toggling GPIO33  (WL_REG_ON - powers up the chip) and monitoring GPIO42 (WL_HOST_WAKE). No toggling happened, though I can bring up the chip and even have Wifi up - unsure if the GPIO42 not coming through is a USI SIP package issue or our board issue. In regards, to sdhci-msm module loading, I thought I saw some timing issues between the MSM driver coming up and the Broadcom FMAC coming up on probe, maybe I am being superstitious. I'll put some comments in.

Reply:
Did you check whether GPIO42 was actually connected to the module? DV3 hardware GPIO42 is DNP. We should double check this on Tuesday.